src/syntax/FStar.Syntax.Resugar.fs

```diff
@@ -98,46 +102,45 @@ let rec resugar_universe (u:S.universe) r: A.term =
 
 let string_to_op s =
   let name_of_op = function
-    | "Amp" -> Some "&"
-    | "At" -> Some "@"
-    | "Plus" -> Some "+"
-    (* TODO : there is some arity information which is lost here ! *)
-    | "Minus" -> Some "-"
-    | "Subtraction" -> Some "-"
-    | "Slash" -> Some "/"
-    | "Less" -> Some "<"
-    | "Equals" -> Some "="
-    | "Greater" -> Some ">"
-    | "Underscore" -> Some "_"
-    | "Bar" -> Some "|"
-    | "Bang" -> Some "!"
-    | "Hat" -> Some "^"
-    | "Percent" -> Some "%"
-    | "Star" -> Some "*"
-    | "Question" -> Some "?"
-    | "Colon" -> Some ":"
+    | "Amp" -> Some ("&", 0)
+    | "At" -> Some ("@", 0)
+    | "Plus" -> Some ("+", 0)
```

Comment:
It is only used to tell "subtraction" from "minus" since they are both mapped back to "-" and args.Length is not sufficient since "minus" might have more than one arg due to added args from typechecker. For other operations, it is not needed because we can compute their arity based on the string it mapped back, therefore, it is set to 0.